perfdash/metrics-downloader.go

```diff
@@ -34,39 +34,34 @@ import (
 	"k8s.io/kubernetes/test/e2e/perftype"
 )
 
-// GoogleGCSDownloaderOptions is an options for GoogleGCSDownloader.
-type GoogleGCSDownloaderOptions struct {
+// DownloaderOptions is an options for Downloader.
+type DownloaderOptions struct {
 	ConfigPaths        []string
 	GithubConfigDirs   []string
 	DefaultBuildsCount int
 	LogsBucket         string
 	LogsPath           string
-	CredentialPath     string
 	// Development-only flag.
 	// Overrides build count from "perfDashBuildsCount" label with DefaultBuildsCount.
 	OverrideBuildCount bool
 }
 
-// GoogleGCSDownloader that gets data about Google results from the GCS repository.
-type GoogleGCSDownloader struct {
-	GoogleGCSBucketUtils *bucketUtil
-	Options              *GoogleGCSDownloaderOptions
+// Downloader that gets data about results from a storage service (GCS) repository.
+type Downloader struct {
+	MetricsBuckets MetricsBucket
+	Options        *DownloaderOptions
 }
 
-// NewGoogleGCSDownloader creates a new GoogleGCSDownloader.
-func NewGoogleGCSDownloader(opt *GoogleGCSDownloaderOptions) (*GoogleGCSDownloader, error) {
-	b, err := newBucketUtil(opt.LogsBucket, opt.LogsPath, opt.CredentialPath)
-	if err != nil {
-		return nil, err
+// NewDownloader creates a new Downloader.
+func NewDownloader(opt *DownloaderOptions, bkt MetricsBucket) *Downloader {
+	return &Downloader{
+		MetricsBuckets: bkt,
+		Options:        opt,
 	}
-	return &GoogleGCSDownloader{
-		GoogleGCSBucketUtils: b,
-		Options:              opt,
-	}, nil
 }
 
 // TODO(random-liu): Only download and update new data each time.
-func (g *GoogleGCSDownloader) getData() (JobToCategoryData, error) {
+func (g *Downloader) getData() (JobToCategoryData, error) {
 	configPaths := make([]string, len(g.Options.ConfigPaths))
 	copy(configPaths, g.Options.ConfigPaths)
 	for _, githubURL := range g.Options.GithubConfigDirs {
```

Comment:
No change required for this PR, but probably a follow up, we can refactor getting all the test names from the GitHub API.

Reply:
I can definitely take a look at that :)